tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -541,18 +563,25 @@ static void stream_released(struct bt_bap_stream *stream)
 
 	audio_conn = &connections[a_stream->conn_id];
 
-	if (audio_conn->unicast_group != NULL) {
+	if (stream->dir == BT_AUDIO_DIR_SINK) {
+		/* Stop send timer */
+		k_work_cancel_delayable(&a_stream->audio_clock_work);
+		k_work_cancel_delayable(&a_stream->audio_send_work);
+	}
+
+	if (audio_conn->cigs[stream->ep->cig_id] != NULL) {
+		/* The unicast group will be deleted only at release of the last stream */
 		LOG_DBG("Deleting unicast group");
 
-		int err = bt_bap_unicast_group_delete(audio_conn->unicast_group);
+		int err = bt_bap_unicast_group_delete(audio_conn->cigs[stream->ep->cig_id]);
```

Comment:
Storing `cigs` in the `audio_conn` may not be the right way to go. CIGs are not related to specific connections, and for e.g. AC 7(ii) you will have a single CIG over 2 ACLs. May need to store the CIGs at a "higher" layer